syntax = "proto3";

package security;

import "models/security/security.proto";
import "models/position/position_filter.proto";

option java_multiple_files = true;
option java_package = "common.request";
option java_outer_classname = "CreateSecurityRequestProtos";

//Use this request to create or update securities. Uniqueness is guaranteed via the UUID.
//Security identifiers do not guarantee uniqueness. As an example a bond ISIN or stock ticker
//may be re-used over time. Therefore if you send 2 requests with the same security identifier
//you will create two securities. In order to avoid duplication you should either re-use the UUID
//when calling the API, in which case an update will be applied. If you do not know the UUID, you
//should first do a search operation.
//
//It is preferred that the client generates the UUID. This will avoid issues in the network leading
//to duplicate securities.
message CreateSecurityRequestProto {
  string object_class = 1;
  string version = 2;

  //A fully formed security object to be created or updated. Validations may be applied
  //before creating. For example creating an equity security with bond fields may be invalid and
  //therefore rejected.
  security.SecurityProto security_input = 20;
}


